lib/uring/uring.mli

```diff
@@ -225,6 +225,92 @@ val splice : 'a t -> src:Unix.file_descr -> dst:Unix.file_descr -> len:int -> 'a
     The operation returns the number of bytes transferred, or 0 for end-of-input.
     The result is [EINVAL] if the file descriptors don't support splicing. *)
 
+module Statx : sig
+  type internal
+  (** An internal type used to pass into {! statx}. *)
+
+  type kind = [
+    | `Unknown
+    | `Fifo
+    | `Character_special
+    | `Directory
+    | `Block_device
+    | `Regular_file
+    | `Symbolic_link
+    | `Socket
+  ]
+
+  type t = {
+    blksize : Int64.t;
+    attributes : Int64.t;
+    nlink : Int64.t;
+    uid : Int64.t;
+    gid : Int64.t;
+    mode : int;
+    ino : Int64.t;
+    size : Optint.Int63.t;
+    blocks : Int64.t;
+    attributes_mask : Int64.t;
+    atime : float;
+    btime : float;
+    ctime : float;
+    mtime : float;
+    rdev : Int64.t;
+    dev : Int64.t;
+    perm : int;
+    kind : kind;
+    mask : Int64.t;
+  } (** See statx(2). *)
+
+  val create : unit -> internal
+  (** Use [create] to make an {! internal} statx struct to pass to {! statx}. *)
+
+  val internal_to_t : internal -> t
```

Comment:
One of the key benefits to statx is that you can retrieve only the values you need, and not have everything else present. So instead of the big type `t`, we could just have `noalloc` accessors that convert the relevant `internal` struct field into an OCaml value.  The `internal` block can be reused across sequential statx calls, and pools of them used for parallel invocations. This has the benefit that a lot of statx calls would have minimal allocation only for the fields that they need.